nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/NuxeoOAuth2Servlet.java

```diff
@@ -196,30 +193,35 @@ else if (!authRequest.getClientId().equals(tokenRequest.getClientId())) {
 
     protected void doPostAuthorizationSubmit(HttpServletRequest request, HttpServletResponse response)
             throws IOException {
-        AuthorizationRequest authRequest = AuthorizationRequest.from(request);
+        String authKeyForm = request.getParameter(AUTHORIZATION_KEY);
+        AuthorizationRequest authRequest = AuthorizationRequest.get(authKeyForm);
+        if (authRequest == null) {
+            handleError(OAuth2Error.INVALID_REQUEST, request, response, null);
+            return;
+        }
+
+        AuthorizationRequest.remove(authRequest.getAuthorizationKey());
+
         OAuth2Error error = authRequest.checkError();
         if (error != null) {
             handleError(error, request, response, authRequest.getRedirectUri());
             return;
         }
 
         String grantAccess = request.getParameter(GRANT_ACCESS_PARAM);
-        String denyAccess = request.getParameter(DENY_ACCESS_PARAM);
 
         // Ensure that the user actually grant access and the authorization key is the correct one
-        String authKeyForm = request.getParameter(AUTHORIZATION_KEY);
-        if ((denyAccess != null || grantAccess == null) || !authRequest.getAuthorizationKey().equals(authKeyForm)) {
+        if (grantAccess == null || !authRequest.getAuthorizationKey().equals(authKeyForm)) {
             handleError(OAuth2Error.ACCESS_DENIED, request, response, authRequest.getRedirectUri());
             return;
         }
 
-        // Save username in request object
-        authRequest.setUsername((String) request.getSession().getAttribute(USERNAME_KEY));
-
+        String authorizationCode = authRequest.getAuthorizationCode();
+        AuthorizationRequest.store(authorizationCode, authRequest);
```

Comment:
I find it difficult to understand why it's an `authorizationCode` here and `authorizationKey` value above.